projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-interface/wpcom-admin-interface.php

```diff
@@ -739,15 +756,28 @@ function wpcom_dismiss_removed_calypso_screen_notice() {
 
 			// If $preferences is not array we log the contents so that we can further debug.
 			if ( ! is_array( $preferences ) && function_exists( 'log2logstash' ) ) {
+				$blog_id = wpcom_get_current_blog_id();
 				log2logstash(
 					array(
 						'feature' => 'wpcom-dismiss-wp-admin-notice',
 						'message' => 'Retrieved a non-array value from Calypso preferences.',
-						'extra'   => wp_json_encode( $preferences ),
+						'extra'   => wp_json_encode(
+							array(
+								'preferences' => $preferences,
+								'user_id'     => get_current_user_id(),
+								'blog_id'     => $blog_id,
+							)
+						),
 					)
 				);
-				// Bail if we can't update the preferences array.
-				wp_die();
+
+				// The expected value when preferences aren't set is an empty string.
+				if ( $preferences !== '' ) {
```

Comment:
Non-blocking: Maybe we should only log this scenario? That way it'd be easier to monitor the cases that we're not handling.

Reply:
Sure